### PR TITLE
Expand How to tune for disk usage

### PR DESCRIPTION
**Pull request by cwurm**
Adding sections on:
- Shard size (20-30 GB)
- Disabling `_source` when possible
- Force Merge
- Shrink

More suggestions welcome.

### CODE REVIEW

File: docs/reference/how-to/disk-usage.asciidoc [hunk 1]
```diff
@@ -136,20 +136,40 @@ PUT index
 // CONSOLE
 
 [float]
+=== Watch your shard size
+
+Larger shards are going to be more efficient at storing data. While it is not possible to give a precise recommendation for how large shards should be, and very large shards come with drawbacks (for example, longer recovery times), it is generally possible to have shard sizes in the 20-30 GB range.
```
Comment (jasontedor) [inline docs/reference/how-to/disk-usage.asciidoc]: I don't think we should specify a size here as it depends on too many factors. With fast replica recovery coming that is another mitigating factor (#22484) to one of the drawbacks that you mention.

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left a comment.

File: docs/reference/how-to/disk-usage.asciidoc (hunk 1 above)
Comment (cwurm) [inline docs/reference/how-to/disk-usage.asciidoc]: @jasontedor Makes sense. Do you think we should mention an upper range, e.g. 50 GB?

Comment (jasontedor) [inline docs/reference/how-to/disk-usage.asciidoc]: It's a good question but I don't think so. We are still fighting the "30 GB" heap recommendation, too many people see that number and think it's the magical number where they should set their heap without enough consideration for all the factors involved. Instead, I think that the verbiage is good but we should avoid enshrining specific numbers.

**Comment by cwurm** [comment]
@jasontedor I've updated the shard size recommendation.

File: docs/reference/how-to/disk-usage.asciidoc [hunk 2]
```diff
@@ -136,20 +136,42 @@ PUT index
...
 === Use `best_compression`
 
 The `_source` and stored fields can easily take a non negligible amount of disk
 space. They can be compressed more aggressively by using the `best_compression`
 <<index-codec,codec>>.
 
 [float]
+=== Force Merge
+
+Elasticsearch stores data in segments. Segments make up a Lucene index - a shard in Elasticsearch. The <<indices-forcemerge,`_forcemerge` API>> can be used to reduce the number of segments per shard. In many cases, the number of segments can be reduced to one per shard by setting `max_num_segments=1`.
```
Comment (jasontedor) [inline docs/reference/how-to/disk-usage.asciidoc]: Perhaps turn this around a bit: Elasticsearch stores data in shards. Shards are Lucene indices and are composed of segments. Segments are the actual files on disk, etc.

File: docs/reference/how-to/disk-usage.asciidoc [hunk 3]
```diff
@@ -136,20 +136,42 @@ PUT index
 // CONSOLE
 
 [float]
+=== Watch your shard size
+
+Larger shards are going to be more efficient at storing data. To increase the size of your shards, you can decrease the number of primary shards in an index by <<indices-create-index,creating indices>> with less primary shards, creating less indices (e.g. by leveraging the <<indices-rollover-index,Rollover API>>), or modifying an existing index using the <<indices-shrink-index,Shrink API>>.
+
+Keep in mind that very large shard sizes come with drawbacks, such as longer recovery times.
```
Comment (jasontedor) [inline docs/reference/how-to/disk-usage.asciidoc]: I do wonder if a comment is in order here about how this applies to *full* recoveries.

**Review by jasontedor — COMMENTED** [review]
I left a few suggestions.

File: docs/reference/how-to/disk-usage.asciidoc (hunk 3 above)
Comment (cwurm) [inline docs/reference/how-to/disk-usage.asciidoc]: Maybe, I'm not sure under which circumstances we'd do a full recovery. Can you suggest a wording?

File: docs/reference/how-to/disk-usage.asciidoc (hunk 2 above)
Comment (cwurm) [inline docs/reference/how-to/disk-usage.asciidoc]: Makes sense, how about: "Indices in Elasticsearch are stored in one or more shards. Each shard is a Lucene index and made up of one or more segments - the actual files on disk. Larger segments are more efficient for storing data. The <<indices-forcemerge,`_forcemerge` API>> can be [...]"

File: docs/reference/how-to/disk-usage.asciidoc (hunk 3 above)
Comment (cwurm) [inline docs/reference/how-to/disk-usage.asciidoc]: @jasontedor ping

Comment (jasontedor) [inline docs/reference/how-to/disk-usage.asciidoc]: Sorry, I've been on vacation. I will resume reviewing when I'm fully back tomorrow.

Comment (jasontedor) [inline docs/reference/how-to/disk-usage.asciidoc]: I don't think we need anything elaborate here, something like: "Keep in mind that large shard sizes come with drawbacks such as long full recovery times."

File: docs/reference/how-to/disk-usage.asciidoc (hunk 2 above)
Comment (jasontedor) [inline docs/reference/how-to/disk-usage.asciidoc]: That sounds good to me.

**Comment by cwurm** [comment]
@jasontedor Incorporated your suggestions, thanks a lot. How does it look?

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**